src/cli.rs

```diff
@@ -0,0 +1,725 @@
+use std::path::{Path, PathBuf};
+use std::time::Duration;
+
+#[cfg(feature = "completions")]
+use anyhow::anyhow;
+use clap::{
+    error::ErrorKind, value_parser, Arg, ArgAction, ArgGroup, ArgMatches, Command, Parser,
+    ValueEnum,
+};
+#[cfg(feature = "completions")]
+use clap_complete::Shell;
+use normpath::PathExt;
+
+use crate::error::print_error;
+use crate::exec::CommandSet;
+use crate::filesystem;
+#[cfg(unix)]
+use crate::filter::OwnerFilter;
+use crate::filter::SizeFilter;
+
+// Type for options that don't have any values, but are used to negate
+// earlier options
+struct Negations;
+
+impl clap::FromArgMatches for Negations {
+    fn from_arg_matches(_: &ArgMatches) -> clap::error::Result<Self> {
+        Ok(Negations)
+    }
+
+    fn update_from_arg_matches(&mut self, _: &ArgMatches) -> clap::error::Result<()> {
+        Ok(())
+    }
+}
+
+impl clap::Args for Negations {
+    fn augment_args(cmd: Command) -> Command {
+        Self::augment_args_for_update(cmd)
+    }
+
+    fn augment_args_for_update(cmd: Command) -> Command {
+        cmd.arg(
+            Arg::new("no_hidden")
+                .action(ArgAction::Count)
+                .long("no-hidden")
+                .overrides_with("hidden")
+                .hide(true)
+                .long_help("Overrides --hidden."),
+        )
+        .arg(
+            Arg::new("ignore")
+                .action(ArgAction::Count)
+                .long("ignore")
+                .overrides_with("no_ignore")
+                .hide(true)
+                .long_help("Overrides --no-ignore."),
+        )
+        .arg(
+            Arg::new("ignore_vcs")
+                .action(ArgAction::Count)
+                .long("ignore-vcs")
+                .overrides_with("no_ignore_vcs")
+                .hide(true)
+                .long_help("Overrides --no-ignore-vcs."),
+        )
+        .arg(
+            Arg::new("relative_path")
+                .action(ArgAction::Count)
+                .long("relative-path")
+                .overrides_with("absolute_path")
+                .hide(true)
+                .long_help("Overrides --absolute-path."),
+        )
+        .arg(
+            Arg::new("no_follow")
+                .action(ArgAction::Count)
+                .long("no-follow")
+                .overrides_with("follow")
+                .hide(true)
+                .long_help("Overrides --follow."),
+        )
+    }
+}
+
+#[derive(Parser)]
+#[command(
+    version,
+    after_help = "Note: `fd -h` prints a short and concise overview while `fd --help` gives all \
+    details.",
```

Comment:
Note: clap has a changed description for `-h` and `--help` (see your diff output) which makes this `after_help` message somewhat superfluous.